src/GafferSceneUI/TranslateTool.cpp

```diff
@@ -111,6 +112,32 @@ bool TranslateTool::affectsHandlesTransform( const Gaffer::Plug *input ) const
 		input == scenePlug()->transformPlug();
 }
 
+void TranslateTool::updateHandles()
+{
+	handles()->setTransform( handlesTransform() );
+
+	for( int i = 0; i < 3; ++i )
+	{
+		V3f handleDirection( 0 );
+		handleDirection[i] = 1.0f;
+		Translation translation = createTranslation( handleDirection );
+		bool editable = true;
+		for( int j = 0; j < 3; ++j )
+		{
+			if( translation.direction[j] != 0.0f )
+			{
+				const ValuePlug *plug = selection().transformPlug->translatePlug()->getChild( j );
```

Comment:
Should `plug` be child `i` instead, defined outside the `j` loop? Right now, all handles are disabled when any of them is `!settable()`.